### DISCUSSION TITLE
Rich Text Editor

### DISCUSSION THREAD

**Discussion by megawubs** [livewire/flux#790] [Feature Requests] [answered]
Our application currently uses the filament tiptap package (https://v2.filamentphp.com/plugins/tiptap). During initial development of our saas product, this package provided all we needed. But soon it became obvious that we needed more. Thus, we forked the package and went our way with it. Now again we find the limitations of the fork and are searching for a replacement or a way to build a Tiptap editor in our own codebase. When you released the editor component in Flux Pro, we were intrigued. Just now, we finished evaluating the component and have a few questions for you to determine if we should wait a bit longer, or dive into building our own.

1. What are your plans for supporting custom plugins? As we have a few already developed.
2. When do you expect the mentions and image features will be released?

**Comment by mauritskorse** [comment]
Have a look at this discussion as well: https://github.com/livewire/flux/issues/746#issuecomment-2508141457

**Comment by joshhanley** [comment, marked as answer]
We have added support for custom extensions in Flux v2.1.3 and added some more hooks in Flux v2.1.6.

Adding custom extensions has been documented here https://fluxui.dev/components/editor#extensions

The new hooks haven't been documented yet, but you can find details here https://github.com/livewire/flux/issues/1490#issuecomment-2845772587
